library/data/recipe/src/main/java/org/quiltmc/qsl/recipe/api/package-info.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * <h2>The Recipe API.</h2>
+ * <p>
+ * <h3>Quick note about vocabulary in the Recipe API:</h3>
+ * <ul>
+ *     <li>A static recipe is a recipe which is registered in the API once and is automatically added to
+ *     the {@linkplain net.minecraft.recipe.RecipeManager recipe manager} when recipes are loaded.</li>
+ *     <li>A dynamic recipe is a recipe which is registered when recipes are loaded.</li>
+ * </ul>
+ * <p>
+ * <h3>{@link org.quiltmc.qsl.recipe.api.RecipeManagerHelper RecipeManagerHelper}</h3>
+ * The {@link org.quiltmc.qsl.recipe.api.RecipeManagerHelper RecipeManagerHelper} is a helper class focused
+ * on the {@link net.minecraft.recipe.RecipeManager}, it allows you to register static and dynamic recipes,
+ * it also allows you to modify, replace, and remove recipes.
+ * <p>
+ * <h3>{@link org.quiltmc.qsl.recipe.api.RecipeLoadingEvents RecipeLoadingEvents}</h3>
+ * Contains some events to register, modify, and remove recipes.
+ * <p>
+ * <h3>When to use the Recipe API?</h3>
+ * This API targets specific use-cases:
+ * <ul>
+ *     <li>Dynamic recipe registration, in the case the recipes cannot be determined at compile-time.</li>
+ *     <li>Modification and removal of recipes.</li>
+ * </ul>
+ * This API is <b>NOT</b> supposed to be used for known at compile-time data-generation, please use the appropriate tools instead.
```

Comment:
Perhaps "recipe API" isn't a specific enough name then.

"Runtime Registry Registration, Revision, and Removal API"
aka RRRR API /j

Reply:
Eh, recipe API will be home to more recipe-related APIs in the future.

Reply:
Could I use the constructors here to make the recipes for a datagen API? or should the datagen API have its own thing?

Reply:
Since the recipe serializers are patched in this module, sure, you can.
Though, Minecraft's datagen stuff for recipes are more advanced since it also includes recipe unlock advancements.

Reply:
Would the datagen api then use a wrapper around this method + add the unlock advancement?

Reply:
Could for non-Vanilla recipes, but that's a choice for the developer.

For recipe datagen there's the `RecipeJsonProvider` interface which has a method to get the recipe serializer and a toJson/serialize method.